packages/google_maps_flutter/google_maps_flutter_android/example/lib/bitmap_registry.dart

```diff
@@ -0,0 +1,169 @@
+const int _numberOfMarkers = 500;
+
+class _BitmapRegistryBody extends StatefulWidget {
+  const _BitmapRegistryBody();
+
+  @override
+  State<_BitmapRegistryBody> createState() => _BitmapRegistryBodyState();
+}
+
+class _BitmapRegistryBodyState extends State<_BitmapRegistryBody> {
+  final Set<Marker> _markers = <Marker>{};
+  int? _registeredBitmapId;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _registerBitmap();
+  }
+
+  @override
+  void dispose() {
+    _unregisterBitmap();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SingleChildScrollView(
+      child: Column(
+        children: <Widget>[
+          AspectRatio(
+            aspectRatio: 2 / 3,
+            child: ExampleGoogleMap(
+              markers: _markers,
+              initialCameraPosition: const CameraPosition(
+                target: LatLng(0, 0),
+                zoom: 1.0,
+              ),
+            ),
+          ),
+          MaterialButton(
+            onPressed: () async {
+              // Add markers to the map with a custom bitmap as the icon.
+              //
+              // To show potential performance issues:
+              // * large original image is used (800x600px, ~330KB)
+              // * bitmap is scaled down to 64x64px
+              // * bitmap is created once and reused for all markers. This
+              // doesn't help much because the bitmap is still sent to the
+              // platform side for each marker.
+              //
+              // Adding many markers may result in a performance hit,
+              // out of memory errors or even app crashes.
+              final BytesMapBitmap bitmap = await _getAssetBitmapDescriptor();
+              _updateMarkers(bitmap);
+            },
+            child: const Text('Add $_numberOfMarkers markers, no registry'),
+          ),
+          MaterialButton(
+            onPressed: _registeredBitmapId == null
+                ? null
+                : () {
+                    // Add markers to the map with a custom bitmap as the icon
+                    // placed in the bitmap registry beforehand.
+                    final RegisteredMapBitmap registeredBitmap =
+                        RegisteredMapBitmap(id: _registeredBitmapId!);
+                    _updateMarkers(registeredBitmap);
+                  },
+            child: const Text('Add $_numberOfMarkers markers using registry'),
+          ),
+        ],
+      ),
+    );
+  }
+
+  /// Register a bitmap in the bitmap registry.
+  Future<void> _registerBitmap() async {
+    if (_registeredBitmapId != null) {
+      return;
+    }
+
+    final BytesMapBitmap bitmap = await _getAssetBitmapDescriptor();
+    const int registeredBitmapId = 1;
+    await GoogleMapsFlutterPlatform.instance.registerBitmap(1, bitmap);
+
+    // If the widget was disposed before the bitmap was registered, unregister
+    // the bitmap.
+    if (!mounted) {
+      _unregisterBitmap();
```

Comment:
this won't never unregister the bitmap, as the _registeredBitmapId is updated after this call
and at the start of _unregisterBitmap there is check:
```dart
    if (_registeredBitmapId == null) {
      return;
    }
```

Reply:
maybe it would be better to call just 
`GoogleMapsFlutterPlatform.instance.unregisterBitmap(registeredBitmapId!);`
here?

(remember to fix this in all packages)

Reply:
Private field is now set when bitmap ID becomes available. Empty `setState()` is called at the end of the method, I think it should do it